internal/datastore/crdb/reader.go

```diff
@@ -94,7 +94,8 @@ func (cr *crdbReader) CountRelationships(ctx context.Context, name string) (int,
 		return 0, err
 	}
 
-	query := cr.addFromToQuery(countRels, cr.schema.RelationshipTableName)
+	index := schema.IndexForFilter(cr.schema, relFilter)
+	query := cr.addFromToQuery(countRels, cr.schema.RelationshipTableName+"@"+index.Name)
```

Comment:
I understand you don't want to enforce index in a bunch of queries, but I'd love to see `addFromToQuery` taking an Index argument. For those `addFromToQuery` where you don't want an index, you could specifically use a `NoIndex` constant for now. This way:
- next time we add new calls, it forces us to think about indexes
- we can easily find callsites without an index by looking for `NoIndex`

Reply:
Added

Reply:
I'm still seeing the same code

Reply:
Where? This is what is on that line now:

```
query := cr.addFromToQuery(countRels, cr.schema.RelationshipTableName, index.Name)
```

Reply:
sorry, was seeing stale code